### PR TITLE
Update registry checks for image auth

### PR DESCRIPTION
**Pull request by mikestoltz**
@containership/backend

### CODE REVIEW

File: lib/follower.js [hunk 1]
```diff
@@ -35,36 +35,33 @@ module.exports = {
...
-                } else if(options.image.split('/').length === 2 && options.image.split('/')[0].includes('.')) {
-                    // example repo name: 823379307507.dkr.ecr.us-east-1.amazonaws.com/testing
-                    registry = options.image.split('/')[0];
+                registries.push({
+                    username: config.organization,
+                    password: config.api_key,
+                    serveraddress: cs_registry_url.host,
+                    auth: ''
+                });
+
```
Comment (normanjoyner) [inline lib/follower.js]: remove whitespace

File: lib/follower.js [hunk 2]
```diff
@@ -35,36 +35,33 @@ module.exports = {
...
+                    registry_domain = 'docker.io';
                 } else{
-                    registry = 'docker.io';
+                    registry_domain = image_parts[0];
                 }
 
-                if(_.has(registries, registry)) {
-                    return async.forEach(registries[registry], function(registry, fn) {
+                return async.each(registries, (registry, fn) => {
+                    if(registry.serveraddress.includes(registry_domain)) {
```
Comment (normanjoyner) [inline lib/follower.js]: can we explicitly check:
```
if(registry.serveraddress === https://${registry_domain} || registry.serveraddress === https://${registry_domain}) {
}
```

File: lib/follower.js [hunk 3]
```diff
@@ -102,13 +99,17 @@ module.exports = {
...
-
                         return fn();
-                    }, pre_pull_middleware_callback);
-                } else{
-                    options.auth.push({});
+                    } else{
+                        return fn();
+                    }
+                }, () => {
+                    if(!options.auth.length) {
```
Comment (normanjoyner) [inline lib/follower.js]: comment

File: lib/follower.js (hunk 2 above)
Comment (mikestoltz) [inline lib/follower.js]: You just wrote the same thing twice here. But no, then it would not work for the case we discussed yesterday

Comment (normanjoyner) [inline lib/follower.js]: updated

Comment (normanjoyner) [inline lib/follower.js]: add regex

**Comment by mikestoltz** [comment]
@normanjoyner updated

**Pull request merged**